presto-main/src/main/java/com/facebook/presto/operator/window/PagesWindowIndex.java

```diff
@@ -80,6 +82,14 @@ public Slice getSlice(int channel, int position)
     }
 
     @Override
+    public Block getBlock(int channel, int position)
+    {
+        BlockBuilder builder = pagesIndex.getType(channel).createBlockBuilder(new BlockBuilderStatus(), 1);
```

Comment:
Actually, I'm not sure if calling `getRegion()` would be a problem here. @erichwang do you know if holding onto regions would be a problem for streaming window operators?

It might be best to do the copy for now, just to be safe, but you could also benchmark if there is any difference in TPC-DS queries. If there's a big difference, maybe we should skip the copy and deal with it later if there's actually a memory issue?